### PR TITLE
3.0/feature/gutenberg new columns block calc

### PR DESCRIPTION
**Pull request by NiclasNorin**

### CODE REVIEW

**Comment by NiclasNorin** [comment]
Task linked: [CU-865d3tkzy Kolumnval 33/67](https://app.clickup.com/t/865d3tkzy)

File: library/Blocks/Columns.php [hunk 1]
```diff
@@ -27,26 +27,37 @@ private function isBlock(string $name, array $object): bool
...
         //Load doc as string
         $doc = new \DOMDocument();
         $doc->loadHTML('<?xml encoding="utf-8" ?>' . $content);
 
         //Get the columns and its contents
-        $result = [];
-
+        $modifiedColumns = [];
+        $index = 0;
         foreach ($doc->getElementsByTagName('*') as $child) {
```
Comment (sebastianthulin) [inline library/Blocks/Columns.php:61]:
```suggestion
        foreach ($doc->getElementsByTagName('*') as $index =>  $child) {
```

File: library/Blocks/Columns.php [hunk 2]
```diff
@@ -27,26 +27,37 @@ private function isBlock(string $name, array $object): bool
...
+    private function processBlockColumns(string $content, array $gridClasses) {
         //Load doc as string
         $doc = new \DOMDocument();
         $doc->loadHTML('<?xml encoding="utf-8" ?>' . $content);
 
         //Get the columns and its contents
-        $result = [];
-
+        $modifiedColumns = [];
+        $index = 0;
```
Comment (sebastianthulin) [inline library/Blocks/Columns.php:60]:
```suggestion

```

File: library/Blocks/Columns.php (hunk 1 above)
Comment (sebastianthulin) [inline library/Blocks/Columns.php:61]:
```suggestion
        $elements = $doc->getElementsByTagName('*'); 
        if(!empty($elements) && is_array($elements)) {
                foreach ($elements as $child) {
```

File: library/Blocks/Columns.php [hunk 3]
```diff
@@ -55,24 +66,62 @@ public function renderBlockColumns(string $content, array $block): string
...
                         [
-                            $gridClass,
+                            $gridClasses[$index],
                             'o-grid-column-block',
                             str_replace('wp-block-column', '', $class),
                         ]
                     )
                 );
+                $modifiedColumns[] = $child->c14n();
+                $index++;
```
Comment (sebastianthulin) [inline library/Blocks/Columns.php:76]:
```suggestion

```

File: library/Blocks/Columns.php [hunk 4]
```diff
@@ -55,24 +66,62 @@ public function renderBlockColumns(string $content, array $block): string
...
+            }
+        }
```
Comment (sebastianthulin) [inline library/Blocks/Columns.php:78]:
```suggestion
                 }
            }
        }
```

**Review by sebastianthulin — CHANGES_REQUESTED** [review]

**Review by sebastianthulin — APPROVED** [review]

**Pull request merged**